riscv/processor.cc

```diff
@@ -543,6 +543,9 @@ void state_t::reset(processor_t* const proc, reg_t max_isa)
   sscsrind_reg_csr_t::sscsrind_reg_csr_t_p vsireg[6];
 
   if (proc->extension_enabled_const(EXT_SMCSRIND)) {
+    miselect = std::make_shared<basic_csr_t>(proc, CSR_MISELECT, 0);
```

Comment:
Can you move the declaration of the `miselect` variable into this scope?  (And also make the analogous improvement for `siselect` and `vsiselect`.)

Reply:
Add a new commit that combines the declaration and initialization of miselect into a single statement  (as well as siselect and vsiselect).